File: .guides/tests/ch-4.js

keyPressedEvent  = null;

$.getScript(window.location.origin + '/public/js/' + window.testEnv.cmd + '.js?_=' + Date.now())
.done(function (script, status) {
  
  createRandomMaze();
    
  var _left = false;
  var _right = false;
  var _up = false;
  var _down = false;
  var _else = false;
    
  if(typeof keyPressedEvent  == 'function') {

    window.showMessage = function (val) {
      _else = true;
    }        
    
    
    player.moveLeft = function () {
      _left = true;
    }

    keyPressedEvent('LEFT');
    
    if(!_left || _else) {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }   
    

    player.moveRight = function () {
      _right = true;
    }

    keyPressedEvent('RIGHT');
    
    if(!_right || _else) {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }   
    

    player.moveUp = function () {
      _up = true;
    }

    keyPressedEvent('UP');

    if(!_up || _else) {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }   
    
    
    player.moveDown = function () {
      _down = true;
    }

    keyPressedEvent('DOWN');

    if(!_down || _else) {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }   
    
    _left = false;
    _right = false;
    _up = false;
    _down = false;    

    keyPressedEvent('C');
    
    if(_left || _right || _up || _down || !_else) {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }   
  }
  else {
    return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
  }
  
  codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.SUCCESS, 'Well done!');    

})
.fail(function (jqxhr, settings, exception) {
  console.log(exception);
  codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, exception.message); 
});

File: .guides/tests/ch-4.py

direction = ''
message = False

def showMessage (val):
  global message
  message = True

def moveLeft():
  global direction
  direction = 'LEFT'

def moveRight():
  global direction
  direction = 'RIGHT'

def moveUp():
  global direction
  direction = 'UP'

def moveDown():
  global direction
  direction = 'DOWN'

try:
  execfile('/home/codio/workspace/public/py/ch-4.py')
  
  keyPressedEvent('LEFT')
  if direction != 'LEFT':
    raise ValueError('keyPressedEvent is not handinling the LEFT keyCode correctly')
  
  keyPressedEvent('RIGHT')
  if direction != 'RIGHT':
    raise ValueError('keyPressedEvent is not handling the RIGHT keyCode correctly')
  
  keyPressedEvent('UP')
  if direction != 'UP':
    raise ValueError('keyPressedEvent is not handling the UP keyCode correctly')
  
  keyPressedEvent('DOWN')
  if direction != 'DOWN':
    raise ValueError('keyPressedEvent is not handling the DOWN keyCode correctly')

  keyPressedEvent('OTHER')
  if message == False:
    raise ValueError('keyPressedEvent is not handling other keyCode eveents correctly')

  print 'well done'
  exit(0)
  
except (IOError, SyntaxError, ValueError) as e:
  print str(e)
  exit(1)
  
print 'Not quite right, try again!'
exit(1)
